saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/serviceprovider/provider/Saml2RelyingPartyRepository.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.saml2.serviceprovider.provider;
+
+/**
+ * Resolves a configured remote provider by entityId or alias
+ */
+public interface Saml2RelyingPartyRepository {
```

Comment:
I think that since this serves `Saml2RelyingPartyRegistration`s, it should be called `Saml2RelyingPartyRegistrationRepository`, similar to how `ClientRegistrationRepository` returns `ClientRegistration`s.

Reply:
Is that because `ClientRepository` would not be intuitive, or because we already picked a naming strategy and we should stick with it?

Reply:
A repository that services `XYZ`s is typically called `XYZRepository`. I think that's driven by convention, yes, but I wouldn't go so far as to say we picked it -- it seems quite prevalent. If `ClientRegistration`s were just called `Client`s then I think the repository would have been called `ClientRepository`. 

As a side note, since `Saml2RelyingPartyRegistration` aligns with `ClientRegistration` in its role in the API, I think suffixing the domain object with "Registration" is reasonable.